pkg/runtime/grpc/grpc-runtime.go

```diff
@@ -516,14 +511,9 @@ All these options should be set at the same time to enable TLS connection`,
 			gadgetNamespace := r.globalParams.Get(ParamGadgetNamespace).AsString()
 			return NewK8SPortFwdConn(ctx, r.restConfig, gadgetNamespace, target, port, timeout)
 		}))
-	} else {
-		newCtx, cancel := context.WithTimeout(dialCtx, timeout)
```

Comment:
It seems like for the `else` case we are losing the timeout.
We would need to keep that even with the new `NextClient`